### PR TITLE
Improve Javac(Method|Type)Binding.getKey()

### LINKED ISSUE #284: [javac] Fix binding `getKey` method implementations
There are quite a few bugs in the binding `getKey` implementations. Fix these bugs, using the `ExternalAnnotations18Test` test suite in order to help locate the bugs and confirm they are fixed.

### PR DESCRIPTION
**Pull request by datho7561**
- fix binding logic for ExternalAnnotations18Test.testBug470666a
- fix binding logic for ExternalAnnotations18Test.testAnnotateMethodTypeParameter2

(posting so people can keep track of progress or want to help, not ready for merge)

Fixes #284

### CODE REVIEW

File: org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/dom/JavacTypeVariableBinding.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package org.eclipse.jdt.internal.javac.dom;
```
Comment (mickaelistria) [inline org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/dom/JavacTypeVariableBinding.java:14]: A copyright/license header is necessary.

**Comment by mickaelistria** [comment]
Don't hesitate to merge any progress even if it's not complete: as long as it doesn't cause a regression or some structural issue, code is better being pushed in the branch.

**Comment by datho7561** [comment]
Okay, there is some cleanup in the code I still need to do, I'll get that done, then squash and merge.

There is still one test that's failing that Rob and I were aiming to fix. I don't yet understand why the binding key is the way it is:

for `package libs; class Lib1 <S extends Throwable, T, U extends List<T>, V> {}`, currently our portion for `U` is:

Ljava/util/List<TT;>;

whereas ECJ gives

Ljava/util/List<Llibs/Lib1;:TT;>;

**Comment by datho7561** [comment]
Going to make sure this doesn't badly break javac in vscode-java, then squash and merge

**Pull request merged**